src/Microsoft.Health.SqlServer/ISqlConnection.cs

```diff
@@ -9,7 +9,7 @@
 
 namespace Microsoft.Health.SqlServer
 {
-    public interface ISqlConnectionFactory
+    public interface ISqlConnection
```

Comment:
>ISqlConnection [](http://example.com/codeflow?start=21&length=14)

maybe offtopic, but I think it was a decent name, and changing to just connection made things less readable.

```using SqlConnection connection = await _sqlConnection.GetSqlConnectionAsync()```
vs 
```using SqlConnection connection = await _sqlConnectionFactory.GetSqlConnectionAsync```

first one is less characters, but it's confusing, why I need to call `_sqlConnection `to create `SqlConnection connection` object.

Reply:
Good point, but a factory.Get() returns an interface that can have multiple implementation.

DI is doing the Factory work. 

This class is AuthenticationType Specific SqlConnection builder. I can name this to ISqlConnectionBuilder and name the other class I introduced as SqlConnectionHelper or something.  Any suggestions?

Reply:
Take a look now, let me know

Reply:
Honestly as long as there are meaningful word after connection, it should be good. Otherwise it blurs between connection itself and object which provide connection.

`ConnectionBuilder` sounds lovely to me.